schema/pgsql/schema.sql

```diff
@@ -0,0 +1,1904 @@
+  CONSTRAINT fk_history_state_history FOREIGN KEY (state_history_id) REFERENCES state_history (id) ON DELETE CASCADE
+);
+
+ALTER TABLE history ALTER COLUMN id SET STORAGE PLAIN;
+ALTER TABLE history ALTER COLUMN environment_id SET STORAGE PLAIN;
+ALTER TABLE history ALTER COLUMN endpoint_id SET STORAGE PLAIN;
+ALTER TABLE history ALTER COLUMN host_id SET STORAGE PLAIN;
+ALTER TABLE history ALTER COLUMN service_id SET STORAGE PLAIN;
+ALTER TABLE history ALTER COLUMN notification_history_id SET STORAGE PLAIN;
+ALTER TABLE history ALTER COLUMN state_history_id SET STORAGE PLAIN;
+ALTER TABLE history ALTER COLUMN downtime_history_id SET STORAGE PLAIN;
+ALTER TABLE history ALTER COLUMN comment_history_id SET STORAGE PLAIN;
+ALTER TABLE history ALTER COLUMN flapping_history_id SET STORAGE PLAIN;
+ALTER TABLE history ALTER COLUMN acknowledgement_history_id SET STORAGE PLAIN;
+
+CREATE INDEX idx_history_event_time ON history(event_time);
+CREATE INDEX idx_history_acknowledgement ON history(acknowledgement_history_id);
+CREATE INDEX idx_history_comment ON history(comment_history_id);
+CREATE INDEX idx_history_downtime ON history(downtime_history_id);
+CREATE INDEX idx_history_flapping ON history(flapping_history_id);
+CREATE INDEX idx_history_notification ON history(notification_history_id);
+CREATE INDEX idx_history_state ON history(state_history_id);
+CREATE INDEX idx_history_host_service_id ON history(host_id, service_id, event_time);
+
+COMMENT ON COLUMN history.id IS 'sha1(environment.name + event_type + x...) given that sha1(environment.name + x...) = *_history_id';
+COMMENT ON COLUMN history.environment_id IS 'environment.id';
+COMMENT ON COLUMN history.endpoint_id IS 'endpoint.id';
+COMMENT ON COLUMN history.host_id IS 'host.id';
+COMMENT ON COLUMN history.service_id IS 'service.id';
+COMMENT ON COLUMN history.notification_history_id IS 'notification_history.id';
+COMMENT ON COLUMN history.state_history_id IS 'state_history.id';
+COMMENT ON COLUMN history.downtime_history_id IS 'downtime_history.downtime_id';
+COMMENT ON COLUMN history.comment_history_id IS 'comment_history.comment_id';
+COMMENT ON COLUMN history.flapping_history_id IS 'flapping_history.id';
+COMMENT ON COLUMN history.acknowledgement_history_id IS 'acknowledgement_history.id';
+
+COMMENT ON INDEX idx_history_event_time IS 'History filtered/ordered by event_time';
+COMMENT ON INDEX idx_history_host_service_id IS 'Host/service history detail filter';
+
+CREATE SEQUENCE icingadb_schema_id_seq;
+
+CREATE TABLE icingadb_schema (
+  id uint NOT NULL DEFAULT nextval('icingadb_schema_id_seq'),
+  version smalluint NOT NULL,
+  timestamp biguint NOT NULL,
+
+  CONSTRAINT pk_icingadb_schema PRIMARY KEY (id)
+);
+
+ALTER SEQUENCE icingadb_schema_id_seq OWNED BY icingadb_schema.id;
+
+INSERT INTO icingadb_schema (version, timestamp)
+  VALUES (2, extract(epoch from now()) * 1000);
```

Comment:
Do we really want to start with version 2 here and try to keep this version number in sync with the MySQL schema? If we'd to, for a change that only affects one schema file (like for example #420), we'd also have to ship an upgrade for the other that does nothing but bump the schema version.

Reply:
Separating the schema versions makes sense to me.